src/contentmanagerside.cpp

```diff
@@ -123,6 +123,13 @@ void ContentManagerSide::setContentManager(ContentManager *contentManager)
                 auto category = item->data(Qt::UserRole).toString();
                 mp_contentManager->setCurrentCategoryFilter(category);
     });
+    connect(this->mp_contentManager, &ContentManager::catalogChanged, [=](bool isLocal) {
+        if (isLocal) {
+            mp_ui->localFileButton->setChecked(true);
+        } else {
+            mp_ui->allFileButton->setChecked(true);
+        }
+    });
```

Comment:
This is a bit odd also.
Buttons should follow the "value of isLocal". It should not be to contentManagerSide to change their state.
Even more, `setChecked`  will trigger [`toggled` event](https://doc.qt.io/qt-6/qabstractbutton.html#toggled) which will in turn change `isLocal` (https://github.com/kiwix/kiwix-desktop/blob/main/src/contentmanagerside.cpp#L15-L18)

Maybe `isLocal` should be a property (with correcty signal/slot associated) and both contentmanager and buttons should react to them.
[QButtonGroup](https://doc.qt.io/qt-6/qbuttongroup.html) may help here for the implementation.

Reply:
I am not exactly sure how I can use "value of isLocal" without a signal and setting state. I've used buttongroup now but hardly changes anything.